### ISSUE TITLE
Getting HTTP Status 405 for Constructing the Access My Data URL

### ISSUE THREAD

**Issue by rahul-chetu** [virencehealth/apiserver.virencehealth.com#129]
Hi,

URL : https://cpsapisandbox.virenceaz.com:9443/demoAPIServer/oauth2/authorize?state=defaultState&scope=openid%20profile%20patient/*.read%20launch/patient&response_type=code%20id_token&redirect_uri=**https://hh-staging.mydomain.com/test**&aud=https%3A%2F%2Fcpsapisandbox.virenceaz.com%3A9443%2FdemoAPIServer&client_id=**MyCLientId**

I am getting the error title "JBWEB000065: HTTP Status 405".

I am working on process to create functionality to post patient's payment details from any remote center using web app. Let me know the specific APIs and flow step to achieve that

**Comment by nkadmane** [comment]
Please follow below link to generate access token:
**https://mydata.athenahealth.com/access-centricity-apis**

$PostPayment operation can be used to add patient's payment details. To get more information on what request body to sent for HTTP POST call, please follow below link and select Posting in left panel.
**https://mydata.athenahealth.com/fhirapidoc**

**Comment by rahul-chetu** [comment]
We are already following the https://mydata.athenahealth.com/access-centricity-apis but as we mentioned previously that we are getting error. 

Please see the previous comment which have request URL with error message and let us know the more details that why we are getting that error.

**Comment by nkadmane** [comment]
We are not able to reproduce this issue. We used the same URL as posted by you in the first comment and replaced the **redirect_uri** and **client_id** values and it worked.

Please check the redirect URL that you constructed to invoke the workflow.

**Comment by rahul-chetu** [comment]
As you mentioned to post payment "$PostPayment operation can be used to add patient's payment details.". 
As per current understanding, we have too first login in your UI as either personal account or work account and need to get patient code by select patient.
![image](https://user-images.githubusercontent.com/61985382/80014152-83b4c800-84ed-11ea-9f91-5d297436df82.png)

In My Application flow, we will already have patient details (spreadsheet). So, we will have own UI to enter required patient details with paid amount. 

So, can we post payment without login work account and without selecting the patient from your UI. Only directly using some APIs.

Please let me know the list of APIs and APIs sample data(if possible ).

**Comment by Amits2070** [comment]
API resources are secured using Azure Active Directory (AAD). Access token needs to be generated to access APIs and selecting personal account or work account is part of generating access token. It gives the authorization code and using that access token is generated.

All these steps are mentioned in https://mydata.athenahealth.com/access-centricity-apis.

Once that is done you can use the generated access token to invoke any APIs are per requirement.

For post payment below API can be used and here is the sample payload:

 https://cpsapisandbox.virenceaz.com:9443/demoAPIServer/fhir/$PostPayment : POST


Request Body:

 <Basic xmlns="http://hl7.org/fhir">

                <extension url="http://hl7.org/fhir/StructureDefinition/ge-posting-batch">

                                <valueInteger value="1169"/>

                </extension>

                <extension url="BatchName">

                                <valueString value="dv-10202009" />

                </extension>

                <extension url="TicketNo">

                                <valueString value="OV003582" />

                </extension>

                <extension url="http://hl7.org/fhir/StructureDefinition/ge-posting-patientVisitId">

                                <valueInteger value="4900"/>

                </extension>

                <extension url="Source">

                                <valueString value="Patient" />

                </extension>

                <extension url="PaymentType">

                                <valueString value="Cash" />

                </extension>

                <extension url="PayerType">

                                <valueString value="Patient" />

                </extension>

                <extension url="Amount">

                                <valueString value="500.00"/>

                </extension>

                <extension url="PayerId">

                                <valueDecimal value="1009843200000000"/>

                </extension>

                <extension url="TransactionDistribution">

                                <valueString value="Yes" />

                </extension>

</Basic>

 

Please replace the correct values for BatchNumber, BatchName, TicketNo, PatientVisitId, Amount and PayerId

**Comment by rahul-chetu** [comment]
Thank you for your response. We understand that we can generate an authorization code for a work account as explained on the developer portal. 

However, we do not want to generate authorization code using UI/ manual steps. We want to generate this authorization code automatically, through an api. We will then use this authorization code to post patient payments (https://cpsapisandbox.virenceaz.com:9443/demoAPIServer/fhir/$PostPayment)

So our question is - can we generate this authorization code automatically through apis?

**Comment by Torin-Shepard-athena** [comment]
Please go [API Resources](https://mydata.athenahealth.com/fhirapidoc) and click on Security to learn more about the authorization services/flows and options. If you need more information, you can send email to athenaInterop@athenahealth.com.

**Comment by KedarGanta** [comment]
@rahul-chetu closing the issue. Please open a new issue if you need additional help.
